### PR TITLE
Test fixes in v2

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: e2e/test/iothub/service/RegistryE2ETests.cs [hunk 1]
```diff
@@ -119,7 +119,13 @@ public async Task DevicesClient_AddDeviceWithTwinWithDeviceCapabilities()
 
             try
             {
-                Device actual = await serviceClient.Devices.GetAsync(deviceId).ConfigureAwait(false);
+                Device actual = null;
```
Comment (drwill-ms) [inline e2e/test/iothub/service/RegistryE2ETests.cs:122]: This test failed for me 2x in a PR build.

File: iothub/device/tests/Pipeline/RetryDelegatingHandlerImplicitOpenTests.cs [hunk 2]
```diff
@@ -140,22 +142,37 @@ public async Task OpenAsyncNextFailedSutIsOpenAndCanBeReopen()
...
-            var tcs = new TaskCompletionSource<int>();
-            tcs.SetCanceled();
-            var nextHandlerMock = Substitute.For<IDelegatingHandler>();
-            nextHandlerMock.OpenAsync(Arg.Any<CancellationToken>()).Returns(tcs.Task);
+            // arrange
 
-            var sut = new RetryDelegatingHandler(contextMock, nextHandlerMock);
-            var cancellationToken = new CancellationToken();
-            await sut.OpenAsync(cancellationToken).ExpectedAsync<TaskCanceledException>().ConfigureAwait(false);
+            var contextMock = new PipelineContext();
```
Comment (drwill-ms) [inline iothub/device/tests/Pipeline/RetryDelegatingHandlerImplicitOpenTests.cs:149]: Let's use Moq for this, and get rid of the task completion source.

File: e2e/test/iothub/service/RegistryE2ETests.cs [hunk 3]
```diff
@@ -119,7 +119,13 @@ public async Task DevicesClient_AddDeviceWithTwinWithDeviceCapabilities()
...
+                do
+                {
+                    // allow some time for the device registry to update the cache
+                    await Task.Delay(50).ConfigureAwait(false);
+                    actual = await serviceClient.Devices.GetAsync(deviceId).ConfigureAwait(false);
+                } while (actual == null);
```
Comment (timtay-microsoft) [inline e2e/test/iothub/service/RegistryE2ETests.cs:128]: Don't we have a "retry this operation" helper method for this kind of thing?

**Review by timtay-microsoft — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]

File: e2e/test/iothub/service/RegistryE2ETests.cs (hunk 3 above)
Comment (drwill-ms) [reply e2e/test/iothub/service/RegistryE2ETests.cs:128]: Yeah, but it takes more lines than this and I think this is more readable.

**Pull request merged**